### PR TITLE
fix: no return value on some case

### PR DESCRIPTION
**Pull request by kegechen**
no return value after asyncCall

Log: undefined behavior
Influence: none
Change-Id: I15a6aec8a8ab57e446f25fbf91c979acde667903

### CODE REVIEW

File: src/kernel/dguiapplicationhelper.cpp [hunk 1]
```diff
@@ -1367,6 +1367,42 @@ void DGuiApplicationHelper::setSingelInstanceInterval(int interval)
...
+    // search all subdirectories
+    const QString xdgDataPath = qgetenv("XDG_DATA_DIRS");
+    auto dataPath = xdgDataPath.split(":");
+    for (const auto &path : dataPath) {
+        QString strManualPath = QStringList {path, "deepin-manual"}.join(QDir::separator());
+
+        QDirIterator it(strManualPath, QDirIterator::Subdirectories);
+        while (it.hasNext()) {
+            QFileInfo file(it.next());
+            if (file.isDir() && file.fileName().contains(appName, Qt::CaseInsensitive)) {
```
Comment (18202781743) [inline src/kernel/dguiapplicationhelper.cpp:1393]: 获取的是Subdirectories，file.isDir()为true?

File: src/kernel/dguiapplicationhelper.cpp [hunk 2]
```diff
@@ -1367,6 +1367,42 @@ void DGuiApplicationHelper::setSingelInstanceInterval(int interval)
...
+                dmanDataExists = true;
+                break;
+            }
+
+            if (file.isDir())
```
Comment (18202781743) [inline src/kernel/dguiapplicationhelper.cpp:1398]: 这个是不是可以去掉？

File: src/kernel/dguiapplicationhelper.cpp [hunk 3]
```diff
@@ -1367,6 +1367,42 @@ void DGuiApplicationHelper::setSingelInstanceInterval(int interval)
...
+    bool dmanDataExists = false;
+    const QString sysPath = qgetenv("PATH");
+    auto binPath = sysPath.split(":");
+    for (const auto &path : binPath) {
+        if (QFile::exists(QStringList {path, "dman"}.join(QDir::separator()))) {
+            dmanBinaryExists = true;
+            break;
+        }
+    }
+
```
Comment (18202781743) [inline src/kernel/dguiapplicationhelper.cpp:1383]: 可以提前return吧

File: src/kernel/dguiapplicationhelper.cpp [hunk 4]
```diff
@@ -1367,6 +1367,42 @@ void DGuiApplicationHelper::setSingelInstanceInterval(int interval)
     DGuiApplicationHelperPrivate::waitTime = interval;
 }
 
+static bool loadManualFromLocalFile()
```
Comment (18202781743) [inline src/kernel/dguiapplicationhelper.cpp:1370]: 函数功能是判断能否加载，而不是加载

**Review by 18202781743 — DISMISSED** [review]

**Review by 18202781743 — APPROVED** [review]

File: src/kernel/dguiapplicationhelper.cpp [hunk 5]
```diff
@@ -1367,6 +1368,43 @@ void DGuiApplicationHelper::setSingelInstanceInterval(int interval)
...
 
+static bool hasLocalManualFile()
+{
+    DCORE_USE_NAMESPACE
+    const QString appName = qApp->applicationName();
+    bool dmanBinaryExists = false;
+    const QString sysPath = qgetenv("PATH");
+    auto binPath = sysPath.split(":");
+    for (const auto &path : binPath) {
+        if (QFile::exists(QStringList {path, "dman"}.join(QDir::separator()))) {
```
Comment (zccrs) [inline src/kernel/dguiapplicationhelper.cpp:1379]: 怎么不直接用 QStandardPaths::findExecutable?

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/kernel/dguiapplicationhelper.cpp (hunk 5 above)
Comment (kegechen) [reply src/kernel/dguiapplicationhelper.cpp:1379]: https://github.com/linuxdeepin/dtkgui/pull/90

**Comment by kegechen** [comment]
https://github.com/linuxdeepin/dtkgui/pull/90

**Pull request closed without merging**